bee-tangle/src/vertices.rs

```diff
@@ -21,6 +24,59 @@ fn equivalent_id(message_id: &MessageId) -> impl Fn(&(MessageId, Vertex)) -> boo
 
 type Table = RwLock<RawTable<(MessageId, Vertex)>>;
 
+pub(crate) struct OccupiedEntry<'a> {
+    // hash: u64,
+    // message_id: Option<MessageId>,
+    elem: Bucket<(MessageId, Vertex)>,
+    table: RwLockWriteGuard<'a, RawTable<(MessageId, Vertex)>>,
+}
+
+impl<'a> OccupiedEntry<'a> {
+    pub(crate) fn into_mut(self) -> RwLockMappedWriteGuard<'a, Vertex> {
+        RwLockWriteGuard::map(self.table, |_| unsafe { &mut self.elem.as_mut().1 })
+    }
+}
+
+pub(crate) struct VacantEntry<'a> {
+    hash: u64,
+    message_id: MessageId,
+    table: RwLockWriteGuard<'a, RawTable<(MessageId, Vertex)>>,
+    hash_builder: &'a DefaultHashBuilder,
+    len: &'a AtomicUsize,
+}
+
+impl<'a> VacantEntry<'a> {
+    pub(crate) fn insert_empty(self) -> RwLockMappedWriteGuard<'a, Vertex> {
+        RwLockWriteGuard::map(self.table, |table| {
+            let entry = table.insert_entry(self.hash, (self.message_id, Vertex::empty()), move |(message_id, _)| {
+                make_hash(&self.hash_builder, message_id)
+            });
+            self.len.fetch_add(1, Ordering::Relaxed);
+            &mut entry.1
+        })
+    }
+}
+
+pub(crate) enum Entry<'a> {
```

Comment:
```suggestion
#[must_use]
pub(crate) enum Entry<'a> {
```

Reply:
In general, I think we should slap `#[must_use]` onto many function in `Tangle` to find places where we might misuse the API at the moment.